cfme/ansible/playbooks.py

```diff
@@ -97,35 +99,26 @@ def is_displayed(self):
         )
 
 
-class PlaybooksCollection(BaseCollection):
+@attr.s
+class PlaybooksCollection(BaseCollection2):
     """Collection object for the :py:class:`Playbook`."""
 
-    def __init__(self, appliance, parent_repository):
-        self.appliance = appliance
-        self.parent = parent_repository
-
-    def instantiate(self, name, repository):
-        return Playbook(self, name, repository)
-
     def all(self):
         view = navigate_to(Server, "AnsiblePlaybooks")
         playbooks = []
         for entity in view.entities.get_all(surf_pages=True):
-            if entity.data["Repository"] == self.parent.name:
-                playbooks.append(
-                    self.instantiate(entity.data["Name"], entity.data["Repository"])
-                )
+            parent = self.filters.get('parent', None)
+            if (parent and entity.data["Repository"] == parent.name) or not parent:
```

Comment:
`parent is None or entity.data["Repository"] == parent.name` and accepting a potential failure if anyone puts in something other than an instance or None